doc/modules/spamtrap.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: doc_modules
+title: Spamtrap module
+---
+# Spamtrap module
+
+With the spamtrap module you can catch spam trapped emails or even a catch-all domain.
+It is intended for learning bayes spam and, if you run your own fuzzy storage, for learning
+a fuzzy flag with a fuzzy weight. You can also add a score for this symbol in the metrics,
+if you only want to tag such emails. You can use this later in other modules like
+force_actions or metadata_exporter. Examples will be given below.
+
+## Spamtrap setup
+
+Spamtrap currently requires redis being configured and used. The idea is to add all your
+spamtrap emails or domains to redis including several settings that you can combine with the
+settings redis module.
+
+### Module settings
+
+Parameters for the spamtrap modules are listed here.
+
+- `action`: You can optionally set an action
+- `symbol`: The name of a symbol that will be inserted, if a match between
+  recipient and a spam trapped email/domain was found. Defaults to 'SPAMTRAP'
+- `score`: The score for this symbol. It defaults to neutral 0.0
+- `learn_fuzzy`: Boolean that enables or disables fuzzy learning. Defaults to
+  'false'
+- `learn_spam`: Boolean that enables or disables bayes spam learning. Defaults
+  to 'false'
+- `fuzzy_flag`: Fuzzy flag, which must match with a defined flag in fuzzy_check
+  for a 'denied' rule
+- `fuzzy_weight`: A weight factor for the fuzzy rule. It defaults to 10.0
+- `key_prefix`: The redis prefix which is used to find spamtrap records. It
+  defaults to 'sptr\_'
+- `map`: You can define a regexp map, which automatically disables redis for
+  this module
+
+Configuration example /etc/rspamd/local.d/spamtrap.conf:
+
+~~~ucl
+action = "no action";
+score = 1.0;
+learn_fuzzy = true;
+learn_spam = true;
+map = file://$LOCAL_CONFDIR/maps.d/spamtrap.map;
+
+enabled = true;
+~~~
+
+An example of a map file is:
+
+~~~text
+/^test@example\.com$/
+/^.\*@catchalldomain.test$/
+~~~
+
+The first is a full email addressm while the second is a catch-all domain.
+
+### Advanced: settings_redis example
+
+The following is an example that you can use for the spamtrap module. It will look
+in redis and collect settings for dealing with spam trapped emails or domains. You
+can place this in /etc/rspamd/rspamd.conf.local:
+
+~~~ucl
+# Return spamtrap e-mail addresses from Redis
+settings_redis {
+        handlers = {
+                SPAMTRAP = <<EOD
+return function(task)
+        local rcpt = task:get_recipients('smtp')
+        if not (rcpt and #rcpt == 1) then
+                return
+        end
+        return 'sptr_' .. rcpt[1]['addr']:lower()
```

Comment:
Perhaps a SPAMTRAP_DOMAIN example would be helpful. It's not immediately clear what the point of this is & it seems to be somewhat conflated with the next point ("adding emails and domains to redis").